sos/policies/distros/redhat.py

```diff
@@ -332,15 +369,38 @@ def upload_sftp(self):
                             " for obtaining SFTP auth token.")
         _token = None
         _user = None
+
+        # We may have a device token already if we attempted
+        # to upload via http but the upload failed. So
+        # lets check first if there isn't one.
```

Comment:
Here we assume http upload is the default, and we can failover from http to sftp only (but not vice versa). That is correct assumption and I dont expect something changes here - just in (the purely speculative) case we change the default to "sftp, if fails then http", we would have to modify this code (or move else-where to upload_http).

ACK as is now, just a comment about hidden assumption the code relies on (and works fine, as I tested).